### ISSUE TITLE
Highlight non-breaking spaces too 

### ISSUE THREAD

**Issue by Kekun** [ebullient/obsidian-show-whitespace-cm6#219]
### 🔖 Feature description

I write in languages using non-breaking spaces, and I need to tell them apart from regular spaces.

### ✔️ Solution

Some apps I use like GNOME Builder and GTranslator show a tiny downward-facing triangle on non-breaking spaces:
![image](https://github.com/user-attachments/assets/a35d72b1-d126-4ab3-89a8-26a2f7e36bd7)
![image](https://github.com/user-attachments/assets/37ad8294-d0f2-4adf-a1da-979ae1cb9109)

### 📝 Additional Context

Show Whitespace don't highlight them at all, not even like spaces (which is good because they aren't regular spaces), but not as their own thing either.

**Comment by ebullient** [comment]
hmm. I can only really work with what codemirror gives me. If you CSS over those spaces in the developer console, do we ahve anything interesting to work with? Can you give me some raw text to work with rather than just screenshots?

**Comment by ebullient** [comment]
I suspect we'd have to play with adding this extension, too: 

https://codemirror.net/docs/ref/#view.highlightSpecialChars
https://github.com/codemirror/view/blob/main/src/special-chars.ts

**Comment by Kekun** [comment]
Here is a file containing some non-breaking spaces: [nbsp.txt](https://github.com/user-attachments/files/18426790/nbsp.txt)

**Comment by Kekun** [comment]
Ah shoot so it's not easily fixable I assume? 🙁 Could instances of `&nbsp;` in strings be cut out and wrapped in a span with a matching style class like for spaces?

**Comment by OnkelTem** [comment]
Yeah, such a bummer non-breaking space chars get replaced by nbsp w/o options and we can't select them via CSS.
